packages/shared/src/graphql/source/common.ts

```diff
@@ -0,0 +1,27 @@
+import { QueryClient } from 'react-query';
+import { generateQueryKey, RequestKey } from '../../lib/query';
+import { SourceMemberFlag, Squad } from '../sources';
+import { LoggedUser } from '../../lib/user';
+
+export const updateFlagsCache = (
+  client: QueryClient,
+  squad: Squad,
+  user: LoggedUser,
+  update: SourceMemberFlag,
+): void => {
+  const queryKey = generateQueryKey(RequestKey.Squad, user, squad?.handle);
+  client.setQueryData<Squad>(queryKey, (oldData) =>
+    oldData
+      ? {
+          ...oldData,
+          currentMember: {
+            ...oldData.currentMember,
+            flags: {
+              ...(oldData.currentMember.flags ?? {}),
+              ...update,
+            },
+          },
+        }
+      : null,
+  );
```

Comment:
I would make it early if and return to not call `setQueryData` at all if no oldData, that is because like this it will actually write data for that query to be `null` vs not existing at all which can mess with initial data checks on the query.

Reply:
Makes sense. Let's do that.

Reply:
But the old data is coming from the `setQueryData`. Do you mean we should fetch first through `getQueryData`?

Reply:
Yeah, that should work.